src/Types/Output/InterfaceType.hack

```diff
@@ -0,0 +1,31 @@
+namespace Slack\GraphQL\Types;
+
+use namespace HH\Lib\{Dict, Vec};
+use namespace Slack\GraphQL;
+
+abstract class InterfaceType extends NamedOutputType implements GraphQL\Introspection\__Type, CompositeType {
+    const type TCoerced = dict<string, mixed>;
+
+    abstract const keyset<string> FIELD_NAMES;
+
+    abstract public function getFieldDefinition(
+        string $field_name
+    ): ?GraphQL\IResolvableFieldDefinition<this::THackType>;
+
+    <<__Override>>
+    abstract public function resolveAsync(
+        this::THackType $value,
+        \Graphpinator\Parser\Field\IHasFieldSet $field,
+        GraphQL\Variables $vars,
+    ): Awaitable<GraphQL\FieldResult<dict<string, mixed>>>;
+
+    <<__Override>>
+    final public function getKind(): GraphQL\Introspection\__TypeKind {
+        return GraphQL\Introspection\__TypeKind::INTERFACE;
+    }
+
+    <<__Override>>
+    final public function getFields(): vec<GraphQL\Introspection\__Field> {
```

Comment:
If we make `CompositeType` an abstract class extending `NamedOutputType`, we could move `getFields()` and the abstract definition of `getFieldDefinition` into it for both `InterfaceType` and `ObjectType`. A very small win?

Reply:
I'm planning to clean up the FieldDefinition stuff (including removing the resolvers from interface FieldDefinitions) but probably better to wait until the introspection/validation PRs are settled first.